### PR TITLE
Utility: add more tests for batch/batchAll/forceBatch

### PR DESCRIPTION
**Pull request by Szegoo**
Because we don't allow the origin to be none for the batch calls we cannot use any of the batch functions with inherents.

Closes: #12374

Polkadot address: 126X27SbhrV19mBFawys3ovkyBS87SGfYwtwa8J2FjHrtbmA

### CODE REVIEW

**Review by xlc — CHANGES_REQUESTED** [review]
Can we have some positive tests.
i.e. batch should work with council origin.

File: frame/utility/src/tests.rs [hunk 1]
```diff
@@ -679,3 +717,92 @@ fn none_origin_does_not_work() {
...
+		);
+	})
+}
+
+#[test]
+fn batch_works_with_council_origin() {
+	new_test_ext().execute_with(|| {
+		let proposal = RuntimeCall::Utility(UtilityCall::batch {
+			calls: vec![call_transfer(1, 5), call_transfer(1, 5)],
+		});
```
Comment (Szegoo) [inline frame/utility/src/tests.rs:778]: @xlc Should I make this test for batchAll and forceBatch also?

Comment (xlc) [reply frame/utility/src/tests.rs:778]: Yes. It is possible to have one working but not the other one.

File: frame/utility/src/tests.rs [hunk 2]
```diff
@@ -679,3 +717,166 @@ fn none_origin_does_not_work() {
...
+			hash,
+			0,
+			proposal_weight,
+			proposal_len
+		));
+
+		System::assert_last_event(RuntimeEvent::Council(pallet_collective::Event::Executed {
+			proposal_hash: hash,
+			result: Err(BadOrigin.into()),
+		}));
```
Comment (Szegoo) [inline frame/utility/src/tests.rs:880]: Tbh I am not sure why is `batch_all` failing with the council.

Comment (xlc) [reply frame/utility/src/tests.rs:880]: council origin simply cannot do transfer so this isn't a good test case
it is just other batch calls never fail

Comment (Szegoo) [reply frame/utility/src/tests.rs]: Hmm, it seems like none of the batch functions work with council origin. I have done multiple tests locally and all of them resulted in the batch tests failing with the council origin(or maybe I didn't test it correctly?). @xlc

File: frame/utility/src/lib.rs [hunk 3]
```diff
@@ -164,13 +164,13 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		/// Send a batch of dispatch calls.
 		///
-		/// May be called from any origin.
+		/// May be called from either the root or a signed origin.
```
Comment (bkchr) [inline frame/utility/src/lib.rs:167]:
```suggestion
		/// May be called from any origin except `None`.
```

File: frame/utility/src/lib.rs [hunk 4]
```diff
@@ -291,13 +291,13 @@ pub mod pallet {
 		/// Send a batch of dispatch calls and atomically execute them.
 		/// The whole transaction will rollback and fail if any of the calls failed.
 		///
-		/// May be called from any origin.
+		/// May be called from either the root or a signed origin.
```
Comment (bkchr) [inline frame/utility/src/lib.rs:294]:
```suggestion
		/// May be called from any origin except `None`.
```

File: frame/utility/src/lib.rs [hunk 5]
```diff
@@ -403,13 +403,13 @@ pub mod pallet {
 		/// Send a batch of dispatch calls.
 		/// Unlike `batch`, it allows errors and won't interrupt.
 		///
-		/// May be called from any origin.
+		/// May be called from either the root or a signed origin.
```
Comment (bkchr) [inline frame/utility/src/lib.rs:406]:
```suggestion
		/// May be called from any origin except `None`.
```

**Review by bkchr — APPROVED** [review]

File: frame/utility/src/tests.rs (hunk 2 above)
Comment (bkchr) [reply frame/utility/src/tests.rs]: You need to change the `TestBaseCallFilter`.

**Comment by Szegoo** [comment]
@xlc PTAL.

File: frame/utility/src/tests.rs [hunk 6]
```diff
@@ -679,3 +760,139 @@ fn none_origin_does_not_work() {
...
+	})
+}
+
+#[test]
+fn batch_all_works_with_council_origin() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(Utility::batch_all(
+			RuntimeOrigin::from(pallet_collective::RawOrigin::Members(3, 3)),
+			vec![RuntimeCall::Democracy(mock_democracy::Call::external_propose_majority {})]
```
Comment (ggwpez) [inline frame/utility/src/tests.rs:895]: Why do you call `Council::propose` in the two other tests but not here?

**Review by ggwpez — APPROVED** [review]

File: frame/utility/src/tests.rs (hunk 6 above)
Comment (Szegoo) [reply frame/utility/src/tests.rs:895]: Because I need to check for events in these two tests cases for which I need the `proposal_hash`.

**Comment by ggwpez** [comment]
bot merge

**Pull request merged**